sql/src/main/java/org/apache/druid/sql/calcite/run/NativeQueryMaker.java

```diff
@@ -122,6 +125,24 @@ public boolean feature(QueryFeature feature)
         );
       }
     }
+    int numFilters = plannerContext.getPlannerConfig().getMaxNumericInFilters();
+    if (numFilters < 1) {
+      throw new CannotBuildQueryException("maxNumericInFilters must be greater than 0");
+    }
+    //special corner case handling for numeric IN filters
```

Comment:
Can you add some comments on why this edge case is worth an extra check here and why the scope of the check is specialized to a particular filter shape?

Reply:
Added comments